crates/sc-consensus-subspace/src/lib.rs

```diff
@@ -125,13 +125,22 @@ pub struct NewSlotInfo {
     pub solution_range: u64,
 }
 
-/// New slot notification with slot information and sender for solution for the slot
+/// New slot notification with slot information and sender for solution for the slot.
 #[derive(Debug, Clone)]
 pub struct NewSlotNotification {
-    /// New slot information
+    /// New slot information.
     pub new_slot_info: NewSlotInfo,
-    /// Sender that can be used to send solutions for the slot
-    pub solution_sender: TracingUnboundedSender<(Solution<FarmerPublicKey>, Vec<u8>)>,
+    /// Sender that can be used to send solutions for the slot.
+    pub solution_sender: TracingUnboundedSender<Solution<FarmerPublicKey>>,
+}
+
+/// Notification with block header hash that needs to be signed and sender for signature.
+#[derive(Debug, Clone)]
+pub struct SignBlockNotification {
```

Comment:
Hmm, could be more straightforward to use the word _unsigned*_ IMO, `subscribe_unsigned_block`, `UnsignedBlockInfo`, etc.

```suggestion
pub struct UnsignedBlockNotification {
```

Reply:
It is not a block, not even a header. The idea of this name was that it is a notification to sign a block and it only contains things necessary to complete the task.

Reply:
How about `BlockSigningNotification`? don't have any other ideas, :( just feel that `SignBlockNotification` is not super intuitive to me at the first glance.